frontend_lib/src/component/Avatar/Avatar.jsx

```diff
@@ -45,44 +46,58 @@ export class Avatar extends React.Component {
   render () {
     const { props } = this
 
-    const generatedColor = this.generateColorFromName(props.publicName)
+    const publicName = props.user.publicName || props.user.public_name
+    const hasAvatar = Object.keys(props.user).includes('hasAvatar') ? props.user.hasAvatar : props.user.has_avatar
+    const letterAvatar = publicName ? this.getTwoLetters(publicName.toUpperCase()) : '?'
+    const sizeAsNumber = parseInt(props.size.replace('px', ''))
+
+    const generatedColor = publicName ? this.generateColorFromName(publicName) : '#000'
     const fontSize = (widthInt => (widthInt / 2) % 2 === 0 ? widthInt : widthInt + 2)(parseInt(props.size)) / 2
 
     return (
       <div
         className={classnames('avatar-wrapper', props.customClass)}
         style={{ ...props.style }}
-        title={props.publicName}
+        title={publicName || props.t('Unknown')}
       >
         <div
           className='avatar'
           data-cy='avatar'
           style={{
             width: props.size,
             height: props.size,
-            borderRadius: props.size,
             backgroundColor: generatedColor,
             fontSize: fontSize
           }}
         >
-          {this.getTwoLetter(props.publicName.toUpperCase())}
+          {hasAvatar ? (
+            <img
+              className='avatar__img'
+              src={`${props.apiUrl}/workspaces/1/files/1/preview/jpg/${sizeAsNumber}x${sizeAsNumber}/${props.filenameInUrl}`}
```

Comment:
Is it normal that it is "fixed" in the space 1 file 1?

Reply:
No, I added this to test without the backend code, I'll change it to reflect the correct URL.